pglock/core.py

```diff
@@ -308,6 +313,13 @@ def inner(*args: _P.args, **kwargs: _P.kwargs) -> _R | None:
                 if self._acquired or self.side_effect != Skip:
                     return func(*args, **kwargs)
 
+        if not self.lock_id and self._func:
+            module = inspect.getmodule(self._func)
+            int_lock_id = _cast_lock_id(f"{module.__name__}.{self._func.__name__}")
+        else:
+            int_lock_id = self.int_lock_id
+        # unwrap to handle other wrapping decorators - such as celery.shared_task
+        utils._unwrap_func(func)._pglock_advisory_id = int_lock_id  # type: ignore[reportFunctionMemberAccess]
```

Comment:
Its also possible to store a list - when function is decorated multiple times, but I decided to keep complexity in check